### PR TITLE
Close Sidebar when clicking outside of it on mobile

### LINKED ISSUE #4751: Make notification sidebar disappear by changing focus to main window
It would be nice to have to notification sidebar disappear when the user clicks on a location outside of the sidebar.

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

closes #4751 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

https://github.com/inmanta/web-console/assets/113331659/c844d558-54c8-4a2d-a83c-bcb47add7a74

### CODE REVIEW

File: src/UI/Root/Components/Header/Header.tsx [hunk 1]
```diff
@@ -30,11 +30,7 @@ export const Header: React.FC<Props> = ({ noEnv, onNotificationsToggle }) => {
     <>
       <Masthead>
         <MastheadToggle>
-          <PageToggleButton
-            variant="plain"
-            aria-label="Main Navigation"
-            id="uncontrolled-nav-toggle"
```
Comment (LukasStordeur) [inline src/UI/Root/Components/Header/Header.tsx:36]: was the id causing problems?

**Comment by LukasStordeur** [comment]
The ticket is about the notification sidebar, not the main-navigation ;)

File: changelogs/unreleased/4571-mobile-sidebar.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-change-type: patch
-description: "Enhancement bringing back functionality to close Sidebar when clicked outside of it on mobile"
+change-type: minor-improvement
```
Comment (LukasStordeur) [inline changelogs/unreleased/4571-mobile-sidebar.yml:1]: patch was correct, you have the template in the changelog folder if you need it. 
the "minor-improvement" is supposed to go in the "section" part :)

File: src/Slices/Notification/UI/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -33,7 +43,21 @@ export const Drawer: React.FC<Props> = ({ onClose, drawerRef }) => {
...
+
+      if (isDrawerOpen && wasTargetOutsideSidebar) {
+        onClose();
+      }
+    };
+    document.addEventListener("click", close);
+    return () => {
+      document.removeEventListener("click", close);
+    };
+  }, [drawerRef, isDrawerOpen, onClose]);
```
Comment (LukasStordeur) [inline src/Slices/Notification/UI/Drawer/Drawer.tsx:59]: can you add a newline after this one?

File: changelogs/unreleased/4571-mobile-sidebar.yml (hunk 2 above)
Comment (matborowczyk) [reply changelogs/unreleased/4571-mobile-sidebar.yml]: Misunderstood :D

**Review by LukasStordeur — APPROVED** [review]

**Pull request closed without merging**